tests/framework/framework.go

```diff
@@ -96,152 +98,34 @@ type Framework struct {
+// flag parsing happens AFTER ginkgo has constructed the entire testing tree. So anything that uses information from flags
+// cannot work when called during test tree construction.
+func NewFramework(prefix string, config ...Config) *Framework {
 	cfg := Config{}
 	if len(config) > 0 {
 		cfg = config[0]
 	}
-	f, err := NewFramework(prefix, cfg)
-	if err != nil {
-		ginkgo.Fail(fmt.Sprintf("failed to create test framework with config %+v: %v", cfg, err))
-	}
-	return f
-}
-
-// NewFramework makes a new framework and sets up the global BeforeEach/AfterEach's.
-// Test run-time flags are parsed and added to the Framework struct.
-func NewFramework(prefix string, config Config) (*Framework, error) {
 	f := &Framework{
-		Config:   config,
+		Config:   cfg,
 		NsPrefix: prefix,
+		Clients:  ClientsInstance,
+		reporter: reporter,
 	}
 
-	// handle run-time flags
-	if !flag.Parsed() {
-		flag.Parse()
-		klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
-		klog.InitFlags(klogFlags)
-		flag.CommandLine.VisitAll(func(f1 *flag.Flag) {
-			f2 := klogFlags.Lookup(f1.Name)
-			if f2 != nil {
-				value := f1.Value.String()
-				f2.Value.Set(value)
-			}
-		})
-
-		fmt.Fprintf(ginkgo.GinkgoWriter, "** Test flags:\n")
-		flag.Visit(func(f *flag.Flag) {
-			fmt.Fprintf(ginkgo.GinkgoWriter, "   %s = %q\n", f.Name, f.Value.String())
-		})
-		fmt.Fprintf(ginkgo.GinkgoWriter, "**\n")
-	}
-
-	f.KubectlPath = *kubectlPath
-	f.OcPath = *ocPath
-	f.CdiInstallNs = *cdiInstallNs
-	f.KubeConfig = *kubeConfig
-	f.Master = *master
-	f.GoCLIPath = *goCLIPath
-	f.SnapshotSCName = *snapshotSCName
-	f.BlockSCName = *blockSCName
-
-	restConfig, err := f.LoadConfig()
-	if err != nil {
-		// Can't use Expect here due this being called outside of an It block, and Expect
-		// requires any calls to it to be inside an It block.
-		return nil, errors.Wrap(err, "ERROR, unable to load RestConfig")
-	}
-	f.RestConfig = restConfig
-	// clients
-	kcs, err := f.GetKubeClient()
-	if err != nil {
-		return nil, errors.Wrap(err, "ERROR, unable to create K8SClient")
-	}
-	f.K8sClient = kcs
-
-	cs, err := f.GetCdiClient()
-	if err != nil {
-		return nil, errors.Wrap(err, "ERROR, unable to create CdiClient")
-	}
-	f.CdiClient = cs
-
-	extcs, err := f.GetExtClient()
-	if err != nil {
-		return nil, errors.Wrap(err, "ERROR, unable to create CsiClient")
-	}
-	f.ExtClient = extcs
-
-	crClient, err := f.GetCrClient()
-	if err != nil {
-		return nil, errors.Wrap(err, "ERROR, unable to create CrClient")
-	}
-	f.CrClient = crClient
-
 	ginkgo.BeforeEach(f.BeforeEach)
 	ginkgo.AfterEach(f.AfterEach)
-	f.reporter = NewKubernetesReporter()
-
-	f.mux.Lock()
-	utils.CacheTestsData(f.K8sClient, f.CdiInstallNs)
-	f.mux.Unlock()
-
-	return f, err
+	return f
 }
 
 // BeforeEach provides a set of operations to run before each test
 func (f *Framework) BeforeEach() {
-	if !f.SkipControllerPodLookup {
```

Comment:
not needed anymore? I cannot find it

Reply:
It was never used, so I figured I might as well remove it while I was cleaning house.

Reply:
So I was wrong, I found a location it is used.